serialization/src/main/scala/sbt/serialization/json/JsonMethods.scala

```diff
@@ -0,0 +1,126 @@
+package sbt.serialization.json
+
+import org.json4s.{
+  JsonInput,
+  StringInput,
+  StreamInput,
+  ReaderInput,
+  FileInput,
+  JValue,
+  JField,
+  JNothing,
+  JBool,
+  JString,
+  JInt,
+  JDecimal,
+  JArray,
+  JObject,
+  JNull,
+  JDouble
+}
+import java.io.File
+import scala.pickling.PicklingException
+import scala.util.Try
+
+/** An implementation of JsonMethods for json4s that uses Jawn and our own toStrings. */
+object JsonMethods extends org.json4s.JsonMethods[JValue] {
+  // Redner doesn't do anything, as we aren't translating to an intermediate format before rendering.
+  override def render(value: JValue): JValue = value
+  // TODO - Write this.
+  override def pretty(d: JValue): String = compact(d)
+  // Compact rendering.
+  override def compact(d: JValue): String = {
+    val buf = new StringBuilder("")
+    import org.json4s._
+    def trimArr(xs: List[JValue]) = xs.filter(_ != JNothing)
+    def trimObj(xs: List[JField]) = xs.filter(_._2 != JNothing)
+    def append(d: JValue): Unit = {
+      d match {
+        case null => buf.append("null")
+        case JBool(true) => buf.append("true")
+        case JBool(false) => buf.append("false")
+        case JDouble(n) => buf.append(n.toString)
+        case JDecimal(n) => buf.append(n.toString)
+        case JInt(n) => buf.append(n.toString)
+        case JNull => buf.append("null")
+        // TODO - better error message
+        case JNothing => sys.error("can't render 'nothing'")
```

Comment:
Most of the render methods have a filter which removes "JNothing"before even getting here.... so....